### ISSUE TITLE
CICD Builds for Windows and Linux

### ISSUE THREAD

**Issue by fire** [virtual-puppet-project/vpuppr#15]
1. Add a Github workflow
2. Take the artifact from the latest 3.x branch on Godot Engine
3. Export the project
4. Combine with openseeface
4. Save as artifact

**Comment by fire** [comment]
Review, use or modify https://github.com/marketplace/actions/godot-export.

**Comment by fire** [comment]
Fetch openseeface.

```
# Whenever a push is made to the master branch then run the job
on: 
  push:
    branches:
      - master

jobs:
  # job id, can be anything
  export_game:
    # Always use ubuntu-latest for this action
    runs-on: ubuntu-latest
    # Job name, can be anything
    name: Export Game Job
    steps:
      # Always include the checkout step so that 
      # your project is available for Godot to export
    - name: checkout
      uses: actions/checkout@v2.3.1
      # Ensure that you get the entire project history
      with:
        fetch-depth: 0
    - name: export game
      # Use latest version (see releases for all versions)
      uses: firebelley/godot-export@v2.8.0
      with:
        # Defining all the required inputs
        # I used the mono version of Godot in this example
        godot_executable_download_url: https://downloads.tuxfamily.org/godotengine/3.3/rc9/mono/Godot_v3.3-rc9_mono_linux_headless_64.zip
        godot_export_templates_download_url: https://downloads.tuxfamily.org/godotengine/3.3/rc9/mono/Godot_v3.3-rc9_mono_export_templates.tpz
        relative_project_path: ./
      env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}```

**Comment by you-win** [comment]
CI builds will be possible once the next 3.x release of Godot is out (current release is 3.3.2 as of this message). Runtime GLTF-loading support is needed and currently I don't want to provide my own build for CI

**Comment by EMCP** [comment]
just curious, without really knowing godot.. is a mac os build out of the question ?

**Comment by you-win** [comment]
It's not out of the question, but I only have an M1 Mac to test with. `onnxruntime`, one of the dependencies for OpenSeeFace, is not officially supported on Apple silicon just yet, and I don't have an Intel-based Mac to test with.

Thus, I don't feel comfortable releasing Mac builds since I cannot test Mac builds on any of my machines.

**Comment by EMCP** [comment]
Understood, I have an Intel Mac but I guess if I had the compilation
instructions I can just.. try it for myself ?

**Comment by you-win** [comment]
Follow the [instructions in the README](https://github.com/you-win/openseeface-gd#building-from-source). In step 8, you will need to compile Godot yourself. I've ninja-edited the compile command, as it was slightly wrong but should be working now.

**Comment by Supermagnum** [comment]
A PPA for Ubuntu Linux would be nifty, if possible.

**Comment by orowith2os** [comment]
Update 8/2/2022 - A flatpak is made, available on flathub under the name of com.github.virtual-puppet-project.vpuppr. The latest releases will be available on flathub beta, for more reasons than just being alpha releases. 
I plan on making it compile from source soon, so ARM64 compatibility can be possible with the flatpak.

**Comment by you-win** [comment]
nightly releases implemented in 22702b95191aa018ff676603acbf1f794ce253aa

extensions have their own cicd pipelines
